core/commands/storage.go

```diff
@@ -387,28 +443,38 @@ the contents and nonce together to produce a final challenge response.`,
 		// previous step should have information with channel id and price
 		chunkInfo, err := ss.GetChunk(chunkHash)
 		if err != nil {
+			if chunkInfo != nil {
+				chunkInfo.SetState(errState)
```

Comment:
On error from client's side, the retry policy should pick up from here (on the retry PR) instead of setting a terminal err state